### DISCUSSION TITLE
how i can enable focus on custom input when i click on field label.

### DISCUSSION THREAD

**Discussion by BilalShah3** [formkit/formkit#1176] [General]
how i can enable focus on custom input when i click on field label.

**Comment by justin-schroeder** [comment]
Make sure you have the `context.id` assigned to the `id` attribute on the input itself. You should end up with markup where the `<label for="{id}">` matches the `<input id="{id}">`.

**Comment by BilalShah3** [comment]
thanks it works.

**Comment by BilalShah3** [comment]
but i am using head less ui listbox select in formkit where i set this id when i place this id on ListBoxButton my select box will open when i click on label
 `<ListboxButton :id="id" class="relative w-full cursor-default rounded-md bg-white py-1.5 pl-3 pr-10 text-left text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 focus:outline-none focus:ring-2 focus:ring-blue-600 sm:text-sm sm:leading-6"></ListboxButton>`
